### DISCUSSION TITLE
Thermostat setpoint

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#237] [Q&A]
I'm trying to change the thermostat setpoint for node 103 to 76 degrees of cooling using this code:
```
let ValueID = {
      "commandClass": 67,
      "endpoint": 0,
      "property": "setpoint",
      "propertyKey": 2
}

let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 103,
        "method": "setValue",
        "params": [ValueID,76]
    }
}
return Message;
```
![Untitled](https://user-images.githubusercontent.com/16625904/179545149-ce07d216-0005-4015-a929-ed109632b9af.png)

I'm getting back an exception:
```
Error: Node 103 is not yet ready to receive commands.

Error: Node 103 is not yet ready to receive commands.
    at NodeCheck (/data/node_modules/node-red-contrib-zwave-js/zwave-js/zwave-js.js:507:12)
    at ValueAPI (/data/node_modules/node-red-contrib-zwave-js/zwave-js/zwave-js.js:1168:5)
    at Init.Input [as _inputCallback] (/data/node_modules/node-red-contrib-zwave-js/zwave-js/zwave-js.js:576:13)
    at /usr/src/node-red/node_modules/@node-red/runtime/lib/nodes/Node.js:210:26
    at Object.trigger (/usr/src/node-red/node_modules/@node-red/util/lib/hooks.js:166:13)
    at Init.Node._emitInput (/usr/src/node-red/node_modules/@node-red/runtime/lib/nodes/Node.js:202:11)
    at Init.Node.emit (/usr/src/node-red/node_modules/@node-red/runtime/lib/nodes/Node.js:186:25)
    at Init.Node.receive (/usr/src/node-red/node_modules/@node-red/runtime/lib/nodes/Node.js:485:10)
    at Immediate.<anonymous> (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/Flow.js:831:52)
    at processImmediate (node:internal/timers:466:21)
```
What does this mean? Is the thermostat R/O? If so, I'm very disappointed.

**Comment by marcus-j-davies** [comment]
It means, the interview has not yet been completed (thumbs up in the UI)
each device has a ready flag, and its usually set after an interview and a successful ping)

**Reply by fabnavigator** [reply]
I just re-intereviewed the node. It said the interview completed, but I know from experience that when it says that it doesn't mean success. It just gave up. Can you fix your mesage to not just sy it completed when it failed?

I see this in the log (set to warning level):
```
2022-07-18T16:10:37.796Z CNTRLR   [Node 103] Timed out while waiting for a response from the node (ZW0201)
```
I included it unsecured to not have to worry about any of the S2 issues I saw previously. In general the node is working. It's sending out updates just as expected.

I re-interviewed again, and it worked this time, and I was able to set the value.

**Reply by fabnavigator** [reply]
Jist to be clear, I had a thumbs up, even though the interview failed. In the debug log I saw that it skipped once it found an error.

**Reply by fabnavigator** [reply]
I don't know wht there are errors. Here is the health check for that node:
![Untitled](https://user-images.githubusercontent.com/16625904/179560821-2a9b0fd4-d282-4c55-a1e2-ba2c92fce541.png)

**Reply by marcus-j-davies** [reply]
Strange, interview Completed is only mentioned if it really was completed - we already do warn of a failed interview.
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/b21cf7a4bb65d78a9ad8a8d25cf0e2607c4f0e2d/zwave-js/zwave-js.js#L2050


> CNTRLR   [Node 103] Timed out while waiting for a response from the node (ZW0201)

is not uncommon, an interview stage is attempted 3 times, if on the 3rd attempt (or maybe 5 - I cant remember)
its at that point we fail the interview (and not mark it as completed)

The thumbs will only be displayed if everything checked out (including a full interview)
The first interview, may have been a dud (for what ever reason), i'll monitor any reports that come in 👍
